lib/smack/Prelude.cpp

```diff
@@ -253,6 +253,20 @@ FuncDecl* truncBvOp(unsigned width, unsigned width2) {
     makeVars(1, type), type2, body, {makeInlineAttr()});
 }
 
+FuncDecl* bswapBvOp(unsigned width) {
+  std::string type = getBvTypeName(width);
```

Comment:
Why is this piece of code not generated directly in `SmackInstGenerator`?
These functions are not calling each other any more, and so I am not sure why we would introduce them in the prelude.

Reply:
@keram88 or @shaobo-he - please take a look at this question and maybe address it (or answer it).

Reply:
> Why is this piece of code not generated directly in `SmackInstGenerator`?
> These functions are not calling each other any more, and so I am not sure why we would introduce them in the prelude.

Do you mean we put the expression into SmackInstGenerator which has the same effect of inlining these functions?

Reply:
Yes. I think @keram88 just did that.

Reply:
Yes. I think @keram88 just did that.